sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -188,14 +195,75 @@ def list_configuration_settings(
                 key="MyKey", label="MyLabel"
             )
         """
-        error_map = {404: ResourceNotFoundError}
-        key_value = await self._impl.get_key_value(
-            key=key,
-            label=label,
-            error_map=error_map,
-            **kwargs
-        )
-        return ConfigurationSetting._from_key_value(key_value)
+        if match_condition == MatchConditions.Unconditionally:
+            error_map = {404: ResourceNotFoundError}
+            key_value = await self._impl.get_key_value(
+                key=key,
+                label=label,
+                error_map=error_map,
+                **kwargs
+            )
+            return ConfigurationSetting._from_key_value(key_value)
+
+        if match_condition == MatchConditions.IfNotModified:
+            if_none_match = quote_etag(etag) if etag else None
+            error_map = {
+                404: ResourceNotFoundError,
+                412: ResourceModifiedError,
+            }
+            key_value = await self._impl.get_key_value(
+                key=key,
+                label=label,
+                if_none_match=if_none_match,
+                error_map=error_map,
+                **kwargs
+            )
+            return ConfigurationSetting._from_key_value(key_value)
+
+        if match_condition == MatchConditions.IfModified:
+            if_none_match = quote_etag(etag) if etag else None
+            error_map = {
+                304: ResourceNotModifiedError,
+                404: ResourceNotFoundError
+            }
+            key_value = await self._impl.get_key_value(
+                key=key,
+                label=label,
+                if_none_match=if_none_match,
+                error_map=error_map,
+                **kwargs
+            )
+            return ConfigurationSetting._from_key_value(key_value)
+
+        if match_condition == MatchConditions.IfPresent:
+            if_match = "*"
+            error_map = {
+                404: ResourceNotFoundError,
+                412: ResourceNotFoundError
+            }
+            key_value = await self._impl.get_key_value(
+                key=key,
+                label=label,
+                if_match=if_match,
+                error_map=error_map,
+                **kwargs
+            )
+            return ConfigurationSetting._from_key_value(key_value)
+
+        if match_condition == MatchConditions.IfPresent:
```

Comment:
Copy pasta? The block above also checked `match_condition == MatchConditions.IfPresent`